inc/xeve_args.h

```diff
@@ -0,0 +1,164 @@
+ * @note It's the user who is in charge of conversion to integer type if it is needed.
+ * @param args
+ * @param [in] keyl key value for an given option
+ * @param [out] val string value for option of a given index
+ *
+ * @retval XEVE_ERR Option not initialized or set
+ * @retval XEVE_ERR_INVALID_ARGUMENT if the option of a given key not exists or one of the arguments is invalid
+ * @retval XEVE_OK Ok
+ *
+ * @attention The function allocates memory for string so user is in charge of releasing memory when no more needed
+ */
+int XEVE_EXPORT xeve_args_get_option_val(ARGS_PARSER * args, char* keyl,  char** val);
+
+/**
+ * @brief Set string value for an option of a given key value
+ *
+ * @note Keep in mind that, there are a lot of options that have numeric values.
+ * @note It's the user who is in charge of conversion from integer type to string if it is needed.
+ *
+ * @param args
+ * @param idx keyl key value for an given option
+ * @param val string value for option of a given key
+ * @retval XEVE_ERROR Option not initialized or set
+ * @retval XEVE_ERR_INVALID_ARGUMENT if the option of a given key not exists or one of the arguments is invalid
+ * @retval XEVE_OK Ok
+ */
+int XEVE_EXPORT xeve_args_set_option_val(ARGS_PARSER * args, char* keyl,  char* val);
+
+/**
+ * @brief Returns string containig help for an option of a given index
+ *
+ * @param [in] args
+ * @param [in] keyl key value for an given option
+ * @param [out] help String contating description for option of given option or NULL string if the option of given long name exists
+ * @retval NULL No option of given long name exists
+ *
+ * @attention The function allocates memory for string so user is in charge of releasing memory when no more needed
+ */
+int XEVE_EXPORT xeve_args_get_option_description(ARGS_PARSER * args, char* keyl, char ** help);
+
+///////////////////////////////////////////////////////////////////////////////
+// HELPERS
+///////////////////////////////////////////////////////////////////////////////
+
+// The following functions must be deleted in the final implementation
+// pomysl nad zmiana nazwy albo usun
+/**
+ * @brief Check whether all mandatory options have been set
+ *
+ * @param args
+ * @param err_arg
+ * @retval XEVE_ERROR Not all mandatory options have been set
+ * @retval XEVE_OK Ok
+ */
+int XEVE_EXPORT xeve_args_check_mandatory(ARGS_PARSER * args, char ** err_arg);
+
+/* iterator */
+/**
+ * @brief Return the next key value in the options container
+ *
+ * The function lets the user iterate through the options container and returns
+ * the next key value after the one passed as an argument.
+ * If the passed key value is NULL returns the first key value from the options container.
+ *
+ * @param args
+ * @param key string containg key value or NULL
+ *
+ * @retval NULL if there is no next key in the options container
+ * @retval key value Ok
+ */
+char* XEVE_EXPORT xeve_args_next_key(ARGS_PARSER * args, char* key);
```

Comment:
Remove this functions

Reply:
xeve_args_get_option_description(ARGS_PARSER * args, char* keyl, char ** help) function needs the key value to provide a description for a given option related to that key. And to get the subsequent keys the 
 xeve_args_next_key() function is needed. It makes it able to get all subsequent keys for all options.

Reply:
As it comes to xeve_args_check_mandatory() it is still used by xeve_app to check whether all mandatory options are set.